src/frequenz/channels/util/_periodic_timer.py

```diff
@@ -0,0 +1,576 @@
+
+    Starting the timer can be delayed if necessary by using `auto_start=False`
+    (for example until we have a running loop). A call to `reset()`, `ready()`,
+    `receive()` or the async iterator interface to await for a new message will
+    start the timer.
+
+    Example:
+        The most common use case is to just do something periodically:
+
+        ```python
+        async for drift in PeriodicTimer(timedelta(seconds=1.0)):
+            print(f"The timer has triggered {drift=}")
+        ```
+
+        But you can also use [Select][frequenz.channels.util.Select] to combine
+        it with other receivers, and even start it (semi) manually:
+
+        ```python
+        timer = PeriodicTimer(timedelta(seconds=1.0), auto_start=False)
+        # Do some other initialization, the timer will start automatically if
+        # a message is awaited (or manually via `reset()`).
+        select = Select(bat_1=receiver1, timer=timer)
+        while await select.ready():
+            if msg := select.bat_1:
+                if val := msg.inner:
+                    process_data(val)
+                else:
+                    logging.warn("battery channel closed")
+            elif drift := select.timer:
+                # Print some regular battery data
+                print(f"Battery is charged at {battery.soc}%")
+                if stop_logging:
+                    timer.stop()
+                elif start_logging:
+                    timer.reset()
+        ```
+
+        For timeouts it might be useful to use `SkipMissedAndDrift` policy, so
+        the timer always gets automatically reset:
+
+        ```python
+        timer = PeriodicTimer(timedelta(seconds=1.0),
+            auto_start=False,
+            missed_tick_policy=SkipMissedAndDrift(),
+        )
+        select = Select(bat_1=receiver1, heavy_process=receiver2, timeout=timer)
+        while await select.ready():
+            if msg := select.bat_1:
+                if val := msg.inner:
+                    process_data(val)
+                    timer.reset()
+                else:
+                    logging.warn("battery channel closed")
+            if msg := select.heavy_process:
+                if val := msg.inner:
+                    do_heavy_processing(val)
+                else:
+                    logging.warn("processing channel closed")
+            elif drift := select.timeout:
+                logging.warn("No data received in time")
+        ```
+
+        In this case `do_heavy_processing` might take 2 seconds, and we don't
+        want our timeout timer to trigger for the missed ticks, and want the
+        next tick to be relative to the time timer was last triggered.
+    """
+
+    def __init__(
+        self,
+        /,
+        interval: timedelta,
+        *,
+        auto_start: bool = True,
+        # We can use an instance here because TriggerAllMissed is immutable
+        missed_tick_policy: MissedTickPolicy = TriggerAllMissed(),
```

Comment:
I think most of our actor use cases would use `SkipMissedAndResync`.  That should probably be the default behaviour.

Reply:
True. Even more, with the skip and drift policy the timer isn't really periodic anymore, I think what we have now is a generic timer. So I propose to rename the class to `Timer`, make the `missed_tick_policy` a required argument and add shortcut constructors for the most common use cases: `Timer.timeout(timeout)` that returns a timer with skip and drift and 0 tolerance and `Timer.periodic(period, skip_missed_ticks=False)`, using trigger all or skip and resync policies as appropriate.